examples/data-objects/clicker/src/index.tsx

```diff
@@ -62,24 +62,19 @@ export class Clicker extends DataObject<{ Events: IClickerEvents; }> {
     }
 
     private setupAgent() {
-        this.taskManager.lockTask(consoleLogTaskId)
-            .then(async () => {
-                console.log(`Picked`);
-                const clickerAgent = new ClickerAgent(this.counter);
-                // Attempt to reacquire the task if we lose it
-                this.taskManager.once("lost", () => {
-                    clickerAgent.stop();
-                    this.setupAgent();
-                });
-                await clickerAgent.run();
-            }).catch(() => {
-                // We're not going to abandon our attempt, so if the promise rejects it probably means we got
-                // disconnected.  So we'll try again once we reconnect.  If it was for some other reason, we'll
-                // give up.
-                if (!this.runtime.connected) {
-                    this.runtime.once("connected", () => { this.setupAgent(); });
-                }
-            });
+        const clickerAgent = new ClickerAgent(this.counter);
+        this.taskManager.subscribeToTask(consoleLogTaskId);
```

Comment:
This is definitely easier to follow.

In the above (replaced) code, I see the attempt to setup agent again, if connection is lost. Could you clarify the guarantees offered by new code:
a. One and only one client attempts tasks?
b. One and only one client attempts and completes the task?
c. At least one client completes the task, even though many can try/attempt ?

Adding extra comments on our examples should be ok, as they will become eventual reference on what some of these DDSes can accomplish.